db/model/Post.ts

```diff
@@ -287,3 +291,223 @@ export const getBlockContentFromSnapshot = async (
 
     return enrichedBlock?.wpApiSnapshot.data?.wpBlock?.content
 }
+
+/*
+ * Get all the related research and writing for a chart
+ */
+export const getRelatedArticles = async (
+    chartId: number
+): Promise<PostReference[] | undefined> => {
+    const relatedPosts: PostReference[] = (
+        await db.knexInstance().raw(
+            `
+        SELECT
+            p.title,
+            p.slug,
+            p.id
+        FROM
+            posts_with_gdoc_publish_status p
+            JOIN posts_links pl ON p.id = pl.sourceId
+            JOIN charts c ON pl.target = c.slug
+            OR pl.target IN (
+                SELECT
+                    cr.slug
+                FROM
+                    chart_slug_redirects cr
+                WHERE
+                    cr.chart_id = c.id
+            )
+        WHERE
+            c.id = ?
+            AND p.status = 'publish'
+            AND p.isGdocPublished = 0
+            AND p.type != 'wp_block'
+            -- note: we are not filtering by linkType to cast of wider net: if a post links to an
+            -- explorer having the same slug as the grapher chart, we want to surface it as
+            -- a "Related research" as it is most likely relevant.
```

Comment:
I don't quite get this comment - can you try to explain to me in different words what you mean?

Reply:
The idea here was to not restrict linkType to "grapher" only, so a post embedding an explorer sharing the same slug as the chart would also be returned (next to posts linking to the chart only). This was more of an edge case optimization which I ended up dropping when I merged references (in the grapher admin) with related posts (in the grapher page) in https://github.com/owid/owid-grapher/pull/3179/commits/a58fea7bb077183f97abbb0d2e97f4bbc337a733.